### PR TITLE
Add KeyFile model

### PR DESCRIPTION
**Pull request by Nurts**
### What was wrong ?
* #24 
* Upload was implemented incorrectly

### How it was fixed ?
* `KeyFile` model was added
*  File is retrieved from `request.FILES`
*  Write happens through the model
*  Keyfiles are saved in `settings.MEDIA_ROOT + 'pretix_attestation_model/keyfiles/'`

### CODE REVIEW

File: pretix_attestation_plugin/views.py [hunk 1]
```diff
@@ -24,24 +21,19 @@ def get_form_kwargs(self):
         return kwargs
 
     def form_valid(self, form):
-        self.write_to_file(form.cleaned_data["key_file_data"])
+        self.write_to_file(self.request.FILES.get('keyfile', None))
```
Comment (kvbik) [inline pretix_attestation_plugin/views.py:24]: you still want to get the file data from the `form.cleaned_data` - the form picks it up from the right places from request - well, as long as the form is properly configured!

File: pretix_attestation_plugin/views.py [hunk 2]
```diff
@@ -24,24 +21,19 @@ def get_form_kwargs(self):
...
         self.save_base_url(form.cleaned_data["base_url"])
         return super().form_valid(form)
 
-    def write_to_file(self, cleaned_data):
-        if(cleaned_data is None):
+    def write_to_file(self, uploaded_file):
+        if(uploaded_file is None):
```
Comment (kvbik) [inline pretix_attestation_plugin/views.py:29]: I do think it is correct to depend on `cleaned_data` which comes from the `Form` that takes care of both things: (a) display the config form, (b) validate the uploaded data!

**Comment by kvbik** [comment]
I believe this direction is correct. The database column can totally work. Now just try to handle the uploaded data the right way using the Form instance (some changes to the Form may be necessary ~ haven't looked too deeply on that).

**Comment by Nurts** [comment]
Hi @kvbik 

> Now just try to handle the uploaded data the right way using the Form instance

I am not sure what you meant by handling it the right way. I thought `FormView`[1] already handles it the right way. Then the storage object, that is specified in `KeyFile` model [2] writes it to specified place (currently to `settings.MEDIA_ROOT + 'pretix_attestation_plugin/keyfiles/'`).

[1] https://github.com/efdevcon/pretix-attestation-placeholder-plugin/blob/main/pretix_attestation_plugin/views.py#L13
[2] https://github.com/efdevcon/pretix-attestation-placeholder-plugin/blob/bf7eb587402f594b160a024a6c55c97c482b3cf3/pretix_attestation_plugin/models.py#L37-L42

**Comment by kvbik** [comment]
You partially do, no worries! The place that is (imho) not correct is `self.write_to_file(self.request.FILES.get('keyfile', None))`, because you are taking raw data from the `request` object. You want to work with validated data and that should come from the `Form` instance - which is the `cleaned_data` variable.

**Comment by kvbik** [comment]
Thx @Nurts - the last commit is exactly what I meant! Let's merge this and test in production :D :D

**Review by ligi — APPROVED** [review]

**Pull request merged**